### ISSUE TITLE
Immutable Map usage

### ISSUE THREAD

**Issue by ncying** [swagger-akka-http/swagger-akka-http#111]
https://github.com/swagger-akka-http/swagger-akka-http/blob/cc2f78e36acdf02d80aad968a8100135f5ba1d24/src/main/scala/com/github/swagger/akka/SwaggerHttpService.scala#L118

filteredSchemas is immutable map, which will cause the same issue as #60

**Comment by pjfanning** [comment]
@ncying I'm not really actively maintaining or using this lib. Feel free to submit a patch and test.

**Comment by pjfanning** [comment]
I don't think this is a bug. The filtered schemas are used privately inside generateSwaggerJson and generateSwaggerYaml.

**Comment by ncying** [comment]
@pjfanning if you override the `unwantedDefinitions` and make it non-empty, it will cause the same issue as #60~
